cleanrl_utils/evals/dqn_eval.py

```diff
@@ -0,0 +1,59 @@
+import random
+from typing import Callable
+
+import gym
+import numpy as np
+import torch
+
+
+def evaluate(
+    model_path: str,
+    make_env: Callable,
+    env_id: str,
+    eval_episodes: int,
+    run_name: str,
+    QNetwork: torch.nn.Module,
+    device: torch.device,
+    epsilon: float = 0.05,
+    capture_video: bool = True,
+):
+    envs = gym.vector.SyncVectorEnv([make_env(env_id, 0, 0, capture_video, run_name)])
+    q_network = QNetwork(envs).to(device)
+    q_network.load_state_dict(torch.load(model_path))
+    q_network.eval()
+
+    obs = envs.reset()
+    episodic_returns = []
+    while len(episodic_returns) < eval_episodes:
+        if random.random() < epsilon:
+            actions = np.array([envs.single_action_space.sample() for _ in range(envs.num_envs)])
+        else:
+            q_values = q_network(torch.Tensor(obs).to(device))
+            actions = torch.argmax(q_values, dim=1).cpu().numpy()
+        next_obs, _, _, infos = envs.step(actions)
+        for info in infos:
+            if "episode" in info.keys():
+                print(f"eval_episode={len(episodic_returns)}, episodic_return={info['episode']['r']}")
+                episodic_returns += [info["episode"]["r"]]
+        obs = next_obs
+
+    return episodic_returns
+
+
+if __name__ == "__main__":
+    from huggingface_hub import hf_hub_download
+
+    from cleanrl.dqn import QNetwork, make_env
+
+    model_path = hf_hub_download(repo_id="cleanrl/CartPole-v1-dqn-seed1", filename="q_network.pth")
```

Comment:
It would be nice to have the ability to pass hf-entity and hf-repo name as an optional parameter if a user wants to evaluate model saved to their own repo.

Reply:
Thank you, Kinal. I was thinking of giving a reasonable preset as follows

```
            repo_name = f"{args.env_id}-{args.exp_name}-seed{args.seed}"
            repo_id = f"{args.hf_entity}/{repo_name}" if args.hf_entity else repo_name
```

and the user, of course, has the flexibility to customize the name of the repo through `exp_name` like

```
python dqn.py --exp-name my-custom-dqn
```

and the repo id will be f"{args.hf_entity}/CartPole-v1-my-custom-dqn-seed1"